src/main/java/me/zeroeightsix/kami/module/modules/misc/HighwayTools.kt

```diff
@@ -0,0 +1,738 @@
+                                }
+                            }
+                        }
+                    } else {
+                        if (cornerBlock.value) {
+                            blockOffsets.add(Pair(relativeDirection(cursor, x / 2, 2 * alterDirection), true))
+                        } else {
+                            if (!(x == buildWidth.value || x == buildWidth.value - 1)) {
+                                blockOffsets.add(Pair(relativeDirection(cursor, x / 2, 2 * alterDirection), true))
+                            }
+                        }
+                        if (rims.value && x == buildWidth.value) {
+                            var c = cursor
+                            for (y in range(1, rimHeight.value + 1)) {
+                                c = c.up()
+                                if (buildWidth.value % 2 != 0) {
+                                    blockOffsets.add(Pair(relativeDirection(c, x / 2, 2 * alterDirection), true))
+                                    blockOffsets.add(Pair(relativeDirection(c, x / 2, 2 * alterDirection * (-1)), true))
+                                } else {
+                                    blockOffsets.add(Pair(relativeDirection(c, x / 2, 2 * alterDirection), true))
+                                    blockOffsets.add(Pair(relativeDirection(c, x / 2 - 1, 2 * alterDirection * (-1)), true))
+                                }
+                            }
+                        }
+                        if (clearSpace.value) {
+                            var c = cursor
+                            for (y in range(1, clearHeight.value)) {
+                                c = c.up()
+                                if (rims.value) {
+                                    if (!((x == buildWidth.value || x == buildWidth.value - 1) && y <= rimHeight.value)) {
+                                        blockOffsets.add(Pair(relativeDirection(c, x / 2, 2 * alterDirection), false))
+                                    }
+                                } else {
+                                    blockOffsets.add(Pair(relativeDirection(c, x / 2, 2 * alterDirection), false))
+                                }
+                            }
+                        }
+                    }
+                    flip = !flip
+                }
+            }
+            Mode.FLAT -> {
+                blockOffsets.add(Pair((b.down().north()), true))
+                blockOffsets.add(Pair((b.down().east()), true))
+                blockOffsets.add(Pair((b.down().south()), true))
+                blockOffsets.add(Pair((b.down().west()), true))
+                blockOffsets.add(Pair((b.down().north().east()), true))
+                blockOffsets.add(Pair((b.down().north().west()), true))
+                blockOffsets.add(Pair((b.down().south().east()), true))
+                blockOffsets.add(Pair((b.down().south().west()), true))
```

Comment:
this could be done differently I think, I'd have to check myself

Reply:
`VectorUtils.getBlockPositionsInArea()`